### PR TITLE
Update Contribution Guide and add Work in Progress

### PR DESCRIPTION
**Pull request by francesperry**
Updated the Contribution Guide with more details on proposals [Beam-566] and make feature branches more visible [Beam-602]

R: @davorbonaci

### CODE REVIEW

**Comment by francesperry** [comment]
(Note that test failures are expected, given there's a TODO to fix up the links once we get the new workflow set up.)

File: contribute/contribution-guide.md [hunk 1]
```diff
@@ -182,13 +216,13 @@ For the `github` remote, add an additional fetch reference, which will cause eve
...
-github	https://github.com/apache/incubator-beam.git (push)
-<b>&lt;username&gt;</b>	git@github.com:<b>&lt;username&gt;</b>/beam.git (fetch)
-<b>&lt;username&gt;</b>	git@github.com:<b>&lt;username&gt;</b>/beam.git (push)</code></pre>
+	$ git remote -v
+	apache	https://git-wip-us.apache.org/repos/asf/incubator-beam.git (fetch)
+	apache	https://git-wip-us.apache.org/repos/asf/incubator-beam.git (push)
+	github	https://github.com/apache/incubator-beam.git (fetch)
+	github	https://github.com/apache/incubator-beam.git (push)
+	<username>	git@github.com:<username>/beam.git (fetch)
+	<username>>	git@github.com:<username>/beam.git (push)
```
Comment (davorbonaci) [inline contribute/contribution-guide.md]: double >?

File: contribute/contribution-guide.md [hunk 2]
```diff
@@ -222,43 +255,109 @@ Please make sure to retain authorship of original commits to give proper credit
...
+
+The directions above assume you are submitting code to the `incubator-beam` repository's `master` branch. In addition, there are a few other locations where code is maintained. Generally these follow the same *engage*-*design*-**code**-**review**-**commit** process as above, with some minor adjustments to commands.
+
+### Feature Branches
+
+Some larger features are developed on a feature branch before being merged into master. In particular, this is often used for initial development of new components like SDKs or runners. 
+
+#### Developing
+
+To contribute to code on a feature branch, use the same process as above, but replace `master` with the [name of the branch](/contribute/work-in-progress/#feature-branches).
```
Comment (davorbonaci) [inline contribute/contribution-guide.md]: rm "to" code

File: contribute/contribution-guide.md [hunk 3]
```diff
@@ -222,43 +255,109 @@ Please make sure to retain authorship of original commits to give proper credit
...
-Done. You can delete the local <code>finish-pr-<b>&lt;pull-request-#&gt;</b></code> branch if you like.
+Done. You can delete the local <code>finish-pr-<pull-request-#></code> branch if you like.
+
+## Special Cases
+
+The directions above assume you are submitting code to the `incubator-beam` repository's `master` branch. In addition, there are a few other locations where code is maintained. Generally these follow the same *engage*-*design*-**code**-**review**-**commit** process as above, with some minor adjustments to commands.
+
+### Feature Branches
+
+Some larger features are developed on a feature branch before being merged into master. In particular, this is often used for initial development of new components like SDKs or runners.
```
Comment (davorbonaci) [inline contribute/contribution-guide.md]: somewhere you have master and elsewhere `master`. Unify?

File: contribute/work-in-progress.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+As mentioned in the [Contribution Guide](/contribute/contribution-guide/), all work in Beam is tracked in our [Apache JIRA](https://issues.apache.org/jira/browse/BEAM). In addition, the following types of work may be of particular interest to the Beam community.
+
+* TOC
+{:toc}
+
+
+## Starter Tasks
+
+The community regular tags good getting started tasks with the label `newbie` or `starter`. Use a quick [JIRA search](https://issues.apache.org/jira/issues?jql=project%20%3D%20BEAM%20AND%20resolution%20%3D%20Unresolved%20AND%20labels%20in%20(newbie%2C%20starter)) to identify ways you can get started [contributing](/contribute/contribution-guide/) to Beam.
```
Comment (davorbonaci) [inline contribute/work-in-progress.md]: Can we settle on one, instead of having both labels?

File: contribute/contribution-guide.md [hunk 5]
```diff
@@ -86,15 +120,15 @@ Clone Beam’s read-only GitHub mirror.
 
 Add your forked repository as an additional Git remote, where you’ll push your changes.
 
-<pre><code>$ git remote add <b>&lt;GitHub_user&gt;</b> git@github.com:<b>&lt;GitHub_user&gt;</b>/incubator-beam.git</code></pre>
+	$ git remote add <GitHub_user> git@github.com:<GitHub_user>/incubator-beam.git
```
Comment (davorbonaci) [inline contribute/contribution-guide.md]: can you double check the rendering of all these formatting changes?

**Comment by francesperry** [comment]
(Waiting on proposal workflow creation: https://issues.apache.org/jira/browse/INFRA-12698 )

File: contribute/contribution-guide.md (hunk 5 above)
Comment (francesperry) [inline contribute/contribution-guide.md]: Good from what I can tell. The downside is we loose the ability to emphasize certain parts of the phrase, but at least we don't have stray html tags ;-)

File: README.md [hunk 6]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (francesperry) [inline README.md]: done.

File: contribute/contribution-guide.md [hunk 7]
```diff
@@ -18,7 +18,7 @@ We use a review-then-commit workflow in Beam for all contributions.
 
 **For larger contributions or those that affect multiple components:**
 
-1. **Engage**: We encourage you to work with the Beam community on the [Apache JIRA issue tracker](https://issues.apache.org/jira/browse/BEAM) and [developer’s mailing list](http://beam.incubator.apache.org/mailing_lists/) to identify good areas for contribution.
+1. **Engage**: We encourage you to work with the Beam community on the [Apache JIRA issue tracker](https://issues.apache.org/jira/browse/BEAM) and [developer’s mailing list](http://beam.incubator.apache.org/use/mailing-lists/) to identify good areas for contribution.
```
Comment (francesperry) [inline contribute/contribution-guide.md]: done

File: contribute/contribution-guide.md [hunk 8]
```diff
@@ -32,27 +32,61 @@ We look forward to working with you!
...
+1. Changes to the Beam programming model, such as new primitive transformations or semantic changes to existing transformations.
+1. Significant changes to how Beam concepts are represented in a given SDK.
+1. Changes that may require coordinated effort across multiple components, such as shared API surfaces or testing infrastructure.
+
+
+### Propose a Design
+
+We suggest using [Google Docs](https://docs.google.com/) for sharing designs that may benefit from diagrams or comments. Please remember to make the document world-commentable and add a link to it from the relevant JIRA issue. You may want to start from this [template](https://docs.google.com/document/d/1qYQPGtabN5-E4MjHsecqqC7PXvJtXvZukPfLXQ8rHJs/edit?usp=sharing).  
+
+All major community discussions and decisions should happen on the dev@ list. So once your design document is ready for broader discussion, begin the conversation:
```
Comment (francesperry) [inline contribute/contribution-guide.md]: done

File: contribute/contribution-guide.md [hunk 9]
```diff
@@ -32,27 +32,61 @@ We look forward to working with you!
...
+
+1. Use the `propose design` button to move the issue into the `under discussion` state.
```
Comment (francesperry) [inline contribute/contribution-guide.md]: done

File: contribute/contribution-guide.md [hunk 10]
```diff
@@ -32,27 +32,61 @@ We look forward to working with you!
...
+1. Update the JIRA issue with a link back to the new [mail thread](https://lists.apache.org/list.html?dev@beam.apache.org).
+1. If there are significant changes or discussions that occur within the design document or JIRA issue, make sure to update the thread on dev@ with an update.
+
+### Conclude the Proposal
+
+If agreement is reached and comments have quiesced:
+
+1. Use the `approved by community` button to move the issue the `in implementation` state.
+1. Make sure the dev@ thread is clear about the result.
+1. Update details in the Design Document and JIRA issue as needed.
```
Comment (francesperry) [inline contribute/contribution-guide.md]: done

File: contribute/contribution-guide.md (hunk 5 above)
Comment (francesperry) [inline contribute/contribution-guide.md]: yup, did. they look generally good and definitely better than before with all the lingering html tags

File: contribute/contribution-guide.md (hunk 1 above)
Comment (francesperry) [inline contribute/contribution-guide.md]: done

File: contribute/contribution-guide.md (hunk 3 above)
Comment (francesperry) [inline contribute/contribution-guide.md]: done

File: contribute/contribution-guide.md (hunk 2 above)
Comment (francesperry) [inline contribute/contribution-guide.md]: done

File: contribute/work-in-progress.md (hunk 4 above)
Comment (francesperry) [inline contribute/work-in-progress.md]: 'starter' wins as it's much more heavily used and also what Flink and Spark use.

File: contribute/work-in-progress.md [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+
+Larger features with multiple active developers may be developed on a [feature branch](/contribute/contribution-guide/#feature-branches) before being merged in the master branch. In particular, this is often used for initial development of new components like SDKs or runners.
+
+Current branches include:
+
+| Feature | Branch | JIRA Component | More Info |
+| ---- | ---- | ---- | ---- |
+| Apache Gearpump Runner | [gearpump-runner](https://github.com/apache/incubator-beam/tree/gearpump-runner) | [runner-gearpump](https://issues.apache.org/jira/browse/BEAM/component/12330829) | [README](https://github.com/apache/incubator-beam/blob/gearpump-runner/runners/gearpump/README.md) |
+| Python SDK | [python-sdk](https://github.com/apache/incubator-beam/tree/python-sdk) | [sdk-py](https://issues.apache.org/jira/browse/BEAM/component/12328910) | [README](https://github.com/apache/incubator-beam/blob/python-sdk/sdks/python/README.md) |
+| Apache Spark 2.0 Runner | [runners-spark2](https://github.com/apache/incubator-beam/tree/runners-spark2) | - | [thread](https://lists.apache.org/thread.html/e38ac4e4914a6cb1b865b1f32a6ca06c2be28ea4aa0f6b18393de66f@%3Cdev.beam.apache.org%3E) |
```
Comment (francesperry) [inline contribute/work-in-progress.md]: Added details on the apex branch (and created a corresponding jira component)

**Comment by francesperry** [comment]
Addressed comments. Pulling the proposal portion out into a later CL (pending infra) and will merge the rest.

**Pull request closed without merging**